src/components/sessions/session-detail-view/AttendanceTable.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useEffect, useState } from "react";
+
+import {
+  Button,
+  Checkbox,
+  createMuiTheme,
+  MuiThemeProvider,
+} from "@material-ui/core";
+import _ from "lodash";
+import MUIDataTable, {
+  MUIDataTableColumn,
+  MUIDataTableOptions,
+} from "mui-datatables";
+
+import { ClassDetailResponse, ClassRequest } from "api/types";
+
+type AttendanceTableRow = {
+  id: string;
+  first_name: string;
+  last_name: string;
+  [key: string]: string;
+};
+
+type AttendanceTableProps = {
+  classObj: ClassDetailResponse;
+  isEditing: boolean;
+  onSubmit: (data: ClassRequest) => void;
+};
+
+const monthNames = [
+  "Jan",
+  "Feb",
+  "Mar",
+  "Apr",
+  "May",
+  "Jun",
+  "Jul",
+  "Aug",
+  "Sep",
+  "Oct",
+  "Nov",
+  "Dec",
+];
+
+const AttendanceTable = ({
+  classObj,
+  isEditing,
+  onSubmit,
+}: AttendanceTableProps) => {
+  const [data, setData] = useState<ClassRequest>(_.cloneDeep(classObj));
+  const [tableRows, setTableRows] = useState<AttendanceTableRow[]>([]);
+
+  const handleCheckboxOnClick = (id: number, date: string) => {
+    const dateIndex = data.attendance.findIndex(
+      (element: { date: string }) => element.date === date
+    );
+    const idIndex = data.attendance[dateIndex].attendees.findIndex(
+      (element) => element === id
+    );
+    const newData: ClassDetailResponse = _.cloneDeep(data);
+    if (idIndex === -1) {
+      newData.attendance[dateIndex].attendees.push(id);
+    } else {
+      newData.attendance[dateIndex].attendees.splice(idIndex, 1);
+    }
+    setData(newData);
+  };
+
+  const getMuiTheme = () =>
+    createMuiTheme({
+      overrides: {
+        MUIDataTableToolbar: {
+          actions: {
+            display: "flex",
+            flexDirection: "row",
+            flex: "initial",
+          },
+        },
+      },
+    });
+
+  const options: MUIDataTableOptions = {
+    responsive: "standard",
+    rowsPerPage: 25,
+    rowsPerPageOptions: [25, 50, 100],
```

Comment:
i think on this table it makes sense not to paginate it (i also don't know how that affects the taking attendance logic since it's based on row index) - let's remove these and add `pagination: false`

Reply:
ohh yes this is a rly good call ty!